src/map/script.cpp

```diff
@@ -27340,6 +27340,56 @@ BUILDIN_FUNC(setdialogpospercent){
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/**
+ * Checks if the character has that permission.
+ * permission_check(<permission>{,<char_id>}) -> <bool>
+ */
+BUILDIN_FUNC(permission_check)
+{
+	map_session_data* sd = nullptr;
+
+	if (!script_charid2sd(3, sd))
+		return SCRIPT_CMD_FAILURE;
+
+	e_pc_permission permission = static_cast<e_pc_permission>(script_getnum(st, 2));
```

Comment:
You cast before verifying the range => undefined behavior for values that are not part of the enum.